### ISSUE TITLE
Switching from nint and nuint to IntPtr for an integer representation looks very bad.

### ISSUE THREAD

**Issue by sergey-zinchenko** [xamarin/xamarin-macios#18673]
After the last macos workload update, all my projects won't compile. All the signatures that utilized `niunt` and `nuint` in the Xamarin's classes now awaits `IntPtr`. This is not a good change for several reasons:

1. IntPtr has no direct interoperability with other integer types. So I can't just compare or subtract `int `and `IntrPtr`, for example. 
2. Everybody uses `IntPtr ` to store native pointers, so starting to use it for arithmetical operations looks wired.
3. All the ancient code won't compile and must be refactored

### Actual Behavior
All the code utilizing `nuint` and `nint `types won't compile now.
For example:

```var window = NSApplication.SharedApplication.MainWindow;
            openPanel.BeginSheet(window, ret =>
            {
                if (ret > 0 && openPanel.Url.Path != null)
                    tcs.SetResult(openPanel.Url.Path);
                else
                    tcs.SetResult(string.Empty);
                openPanel.EndSheet(window);
            });
```
  
Failed compile because BeginSheet awaits for `Action<IntPtr>` and IntPtr has no `>` operator.
            
`var nsResponse = new NSHttpUrlResponse(urlSchemeTask.Request.Url, response.StatusCode, "HTTP/1.1", headers);  `

won't compile as signature become ` public NSHttpUrlResponse(
      NSUrl url,
      string mimetype,
      IntPtr expectedContentLength,
      string? textEncodingName);` . This type of change is the most confusing because there are such a types as `NSNumber` and `NSString` and you can try to use `NSNumber.Handle` `NSString.Handle`, which are IntPtrs. 
       
            
A lot of other code will not compile just because all the places where `nuint` or `nint` integer types was used previously now should use IntPtr. While IntPtr does not have direct interoperability with common integer types. 

### Expected Behavior

Work with the integer type normally as intended to be intuitive.

### Environment

tvos                       16.4.7090/7.0.100      VS 17.7.34003.232
macos                      13.3.7090/7.0.100      VS 17.7.34003.232
maui-windows               7.0.92/7.0.100         VS 17.7.34003.232
maui-maccatalyst           7.0.92/7.0.100         VS 17.7.34003.232
maccatalyst                16.4.7090/7.0.100      VS 17.7.34003.232
maui-ios                   7.0.92/7.0.100         VS 17.7.34003.232
ios                        16.4.7090/7.0.100      VS 17.7.34003.232

**Comment by rolfbjarne** [comment]
This sample code works fine for me:

```cs
NSOpenPanel openPanel = null;
NSWindow window = null;
openPanel.BeginSheet(window, ret =>
{
	Console.WriteLine (ret > 0);
});
```

So can you please attach a complete test project + a binlog (https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) so that we can reproduce the problem you're seeing.

**Comment by filipnavara** [comment]
The reasoning in the original post is flawed. It's not that `nint`/`nuint` was replaced by `IntPtr`/`UIntPtr`. It's something that C# 9 did - they are literally the same types, and they do support arithmetic. When the iOS/macOS workload was added in .NET 6 it made the breaking change to use the C# 9 types instead of its own. From the binary compatibility standpoint it's a breaking change, but from the C# standpoint the types are still referred to as `n[u]int` and they do support arithmetic operations.

**Comment by sergey-zinchenko** [comment]
Hi all! Looks like, my installation of the workload or local package cache was corrupted or malfunctioned. All applications started to compile after I executed `dotnet workload update` next time. After investigating and introspecting related assemblies, it turned out that signatures change a bit since the code wants to compile. I have made two screenshots :
First, the origianl situation where I had compilation errors:
![image](https://github.com/xamarin/xamarin-macios/assets/5931885/3c51cefe-9e36-471a-9d2d-b23197b2661f)
And the second one, where all become normal again:
![image](https://github.com/xamarin/xamarin-macios/assets/5931885/2c99122c-5587-4ab2-a280-26f899d6f761)

There is an additinal attribute after the update that was missed previously.
I am not sure whether it is the root cause but it  is the only thing that I have found.
